packages/credentials-interface/src/utils.ts

```diff
@@ -0,0 +1,100 @@
+import { Credential, VerifiableCredential } from './credentials.types';
+import { Presentation } from './presentation.types';
+
+function isObject(obj: unknown): obj is object {
+  return typeof obj === 'object' && obj !== null;
+}
+
+function hasOwnProperty<X extends object, Y extends PropertyKey>(
+  obj: X,
+  key: Y
+): obj is X & Record<Y, unknown> {
+  return Object.prototype.hasOwnProperty.call(obj, key);
+}
+
+function hasRequiredProperties(
+  obj: object,
+  requiredProperties: string[]
+): boolean {
+  return requiredProperties
+    .map((key) => hasOwnProperty(obj, key))
+    .every(Boolean);
+}
+
+function hasValidProof(proof: unknown): boolean {
+  const requiredProofProperties = [
+    'verificationMethod',
+    'proofPurpose',
+    'created',
+    'type',
+  ];
+
+  if (!isObject(proof)) return false;
+  if (!hasRequiredProperties(proof, requiredProofProperties)) return false;
+  if (!hasOwnProperty(proof, 'jws') && !hasOwnProperty(proof, 'proofValue')) {
+    return false;
+  }
+
+  return true;
+}
+
+export function isCredential(
```

Comment:
This can be done more reliable with https://www.npmjs.com/package/ts-interface-checker

Reply:
Oh, sorry, it was asked already

Reply:
We have very simple type checking, do we really need library to do that?

Reply:
Yes, it is completely optional and perhaps should be moved to separate task. But from my point of view it is worth efforts as it reduces amount of code, excludes errors and updates checker when type updates

Reply:
Okay, we can discuss this later on with other team